src/platform_impl/linux/kms/window.rs

```diff
@@ -0,0 +1,488 @@
+use std::{collections::VecDeque, os::unix::prelude::AsRawFd, sync::Arc};
+
+use super::{Card, MODE};
+use drm::control::*;
+use parking_lot::Mutex;
+
+use raw_window_handle::{DrmDisplayHandle, RawDisplayHandle, RawWindowHandle};
+
+use crate::error;
+
+use crate::{
+    dpi::{PhysicalPosition, PhysicalSize, Position, Size},
+    error::{ExternalError, NotSupportedError},
+    platform_impl::{self, Fullscreen, MonitorHandle, VideoMode},
+    window::{
+        CursorGrabMode, CursorIcon, ImePurpose, ResizeDirection, Theme, WindowAttributes,
+        WindowButtons, WindowId,
+    },
+};
+
+pub struct Window {
+    connector: connector::Info,
+    crtc: crtc::Info,
+    ping: calloop::ping::Ping,
+    plane: plane::Handle,
+    cursor: Arc<Mutex<PhysicalPosition<f64>>>,
+    card: Card,
+}
+
+fn find_prop_id<T: ResourceHandle>(
+    card: &Card,
+    handle: T,
+    name: &'static str,
+) -> Option<property::Handle> {
+    let props = card.get_properties(handle).ok()?;
+    let (ids, _vals) = props.as_props_and_values();
+    ids.iter()
+        .find(|&id| {
+            let info = card.get_property(*id).unwrap();
+            info.name().to_str().map(|x| x == name).unwrap_or(false)
+        })
+        .cloned()
+}
+
+macro_rules! find_prop_id {
+    ($id_handle_1:expr,$handle:expr,$prop_name:literal) => {
+        find_prop_id(&$id_handle_1, $handle, $prop_name).ok_or_else(|| {
+            os_error!(platform_impl::OsError::KmsMisc(concat!(
+                "could not get ",
+                $prop_name
+            )))
+        })
+    };
+}
+
+macro_rules! add_property {
+    ($atomic_req:expr,$handle:expr,$id_handle_1:expr,$prop_name:literal,$property:expr,) => {
+        $atomic_req.add_property(
+            $handle,
+            find_prop_id!(&$id_handle_1, $handle, $prop_name)?,
+            $property,
+        );
+    };
+}
+
+pub(crate) fn winid() -> WindowId {
+    // there is only one window
+    0.into()
+}
+
+impl Window {
+    pub fn new<T>(
+        event_loop_window_target: &super::event_loop::EventLoopWindowTarget<T>,
+        _attributes: WindowAttributes,
+        _platform_attributes: platform_impl::PlatformSpecificWindowBuilderAttributes,
+    ) -> Result<Self, error::OsError> {
```

Comment:
Might be worth considering #33 here. Maybe, if a window has already been created, it's better to just return an error instead of returning another window?